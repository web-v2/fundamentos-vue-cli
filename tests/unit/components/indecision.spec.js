import { shallowMount } from '@vue/test-utils';
import Indecision from '@/components/Indecision.vue';

describe('Indecision Component', () => {
    let wrapper;
    let clgSpy;
    beforeEach(() => {
        wrapper = shallowMount(Indecision);
        clgSpy = jest.spyOn(console, 'log')
    });
    test('Debe de hacer match con el snapshot', () => {        
        expect( wrapper.html() ).toMatchSnapshot();
    });
    test('Escribir en el input no debe disparar nada clg ', async() => {
        const input = wrapper.find('input');
        await input.setValue('Hello World');

        expect( clgSpy ).toHaveBeenCalledTimes(1);
        
    });
    test('Escribir el simbolo de "?" debe disparar el fetch ', () => {
        
    });
    test('Prueba el getAnswer ', () => {
        
    });
    test('Prueba en getAnswer - Fallo en el API', () => {
        
    });
});
