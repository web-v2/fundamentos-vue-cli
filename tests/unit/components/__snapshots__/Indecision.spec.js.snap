// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indecision Component Debe de hacer match con el snapshot 1`] = `
<div>
  <!--v-if-->
  <div class="bg-dark"></div>
  <div class="indecision-container"><input type="text" name="text" placeholder="Hazme una pregunta">
    <p>Recuerda terminar con un signo de interrogaci√≥n (?)</p>
    <!--v-if-->
  </div><!-- Fin indecision-container -->
  <div>
    <h2>Desarrollado Por: Samir Vergara V</h2>
  </div>
</div>
`;
